de-lib/src/main/java/org/iplantc/de/apps/client/views/tools/ManageToolsViewToolbarImpl.java

```diff
@@ -16,42 +23,198 @@
+
     }
-    
+
+    @UiHandler("edit")
+    void onEditClicked(SelectionEvent<Item> event) {
+
+    }
+
+    @UiHandler("delete")
+    void onDeleteClicked(SelectionEvent<Item> event) {
+
+    }
+
+    @UiHandler("useInApp")
+    void onUseInAppClicked(SelectionEvent<Item> event) {
+
+    }
+
+    @UiHandler("shareCollab")
+    void onShareCollabClicked(SelectionEvent<Item> event) {
+
+    }
+
+    @UiHandler("sharePublic")
+    void onSharePublicClicked(SelectionEvent<Item> event) {
+
+    }
+
+    @UiHandler("refreshButton")
+    void onRefresh(SelectEvent event) {
+        fireEvent(new RefreshToolsSelectedEvent());
+    }
+
+    @Override
+    public HandlerRegistration addBeforeToolSearchEventHandler(BeforeToolSearchEvent.BeforeToolSearchEventHandler handler) {
+        return addHandler(handler, BeforeToolSearchEvent.TYPE);
+    }
+
+    @Override
+    public HandlerRegistration addToolSearchResultLoadEventHandler(ToolSearchResultLoadEvent.ToolSearchResultLoadEventHandler handler) {
+        return addHandler(handler, ToolSearchResultLoadEvent.TYPE);
+    }
+
+
+    @Override
+    public HandlerRegistration addRefreshToolsSelectedEventHandler(RefreshToolsSelectedEvent.RefreshToolsSelectedEventHandler handler) {
+        return addHandler(handler, RefreshToolsSelectedEvent.TYPE);
+    }
+
+    @Override
+    public void onToolSelectionChanged(ToolSelectionChangedEvent event) {
+        setButtonState(event.getToolSelection());
+    }
+
+    private void setButtonState(List<Tool> tools) {
+        //view support single selection only
+        if (tools != null) {
+            switch (tools.size()) {
+                case 0:
+                      addTool.setEnabled(true);
+                      requestTool.setEnabled(true);
+                      edit.setEnabled(false);
+                      delete.setEnabled(false);
+                      useInApp.setEnabled(false);
+                      shareMenuButton.setEnabled(false);
+                    break;
+                case 1:
+                    Tool selection = tools.get(0);
+                    addTool.setEnabled(true);
+                    requestTool.setEnabled(true);
+                    edit.setEnabled(isEditable(selection));
+                    delete.setEnabled(isOwner(selection));
+                    useInApp.setEnabled(true);
+                    shareMenuButton.setEnabled(isOwner(selection));
+                    shareCollab.setEnabled(isOwner(selection));
+                    sharePublic.setEnabled(isOwner(selection));
+                    break;
+                default:
+                    addTool.setEnabled(true);
+                    requestTool.setEnabled(true);
+                    edit.setEnabled(false);
+                    delete.setEnabled(false);
+                    useInApp.setEnabled(false);
+                    shareMenuButton.setEnabled(false);
+                    shareCollab.setEnabled(false);
+                    sharePublic.setEnabled(false);
```

Comment:
For the default case, if a user has multiple tools selected, I think it might be nice to enable them to delete multiple tools at once instead of one at a time.
Also, we enable multiple apps/data to be shared at a time, we might want to do the same with tools.